.github/workflows/build_and_test.yaml

```diff
@@ -30,4 +30,9 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
           cache: true
       - name: test
-        run: make test
+        run: go test ./... -race -coverprofile=coverage.xml -covermode=atomic
+      - name: Upload coverage to Codecov
+        run: |
+          curl -Os https://uploader.codecov.io/latest/linux/codecov
+          chmod +x codecov
+          ./codecov
```

Comment:
:wave: I just stumbled upon this PR and thought I would mention that Codecov has a Github Action you can use for this if that would be something you want, e.g.:

```
    - name: Upload coverage to Codecov
      uses: codecov/codecov-action@v3
      with:
        token: ${{ secrets.CODECOV_TOKEN }} # optional
        files: ./coverage.xml
        fail_ci_if_error: true
        verbose: true
```

The documentation for it is [here](https://github.com/codecov/codecov-action/tree/master#usage).

Reply:
thanks for your CR, it's a good suggestion. 
i accept it, cc @shaneutt . thanks.